test/integration/PermissionTable.test.ts

```diff
@@ -0,0 +1,218 @@
+import { AccessMode as AM } from '../../src/authorization/permissions/Permissions';
+import { BasicRepresentation } from '../../src/http/representation/BasicRepresentation';
+import type { App } from '../../src/init/App';
+import type { ResourceStore } from '../../src/storage/ResourceStore';
+import { TEXT_TURTLE } from '../../src/util/ContentTypes';
+import { ensureTrailingSlash, joinUrl } from '../../src/util/PathUtil';
+import { AclHelper } from '../util/AclHelper';
+import { getPort } from '../util/Util';
+import {
+  getDefaultVariables,
+  getPresetConfigPath,
+  getTestConfigPath,
+  getTestFolder,
+  instantiateFromConfig, removeFolder,
+} from './Config';
+
+const DEFAULT_BODY = `@prefix solid: <http://www.w3.org/ns/solid/terms#>.
+@prefix ex: <http://www.example.org/terms#>.
+
+ex:custom ex:givenName "Claudia".`;
+
+const INSERT = `@prefix solid: <http://www.w3.org/ns/solid/terms#>.
+@prefix ex: <http://www.example.org/terms#>.
+_:patch a solid:InsertDeletePatch;
+  solid:inserts { ex:custom ex:givenName "Alex". }.`;
+
+const DELETE = `@prefix solid: <http://www.w3.org/ns/solid/terms#>.
+@prefix ex: <http://www.example.org/terms#>.
+
+_:rename a solid:InsertDeletePatch;
+  solid:deletes { ex:custom ex:givenName "Claudia". }.`;
+
+const N3 = 'text/n3';
+const TXT = 'text/plain';
+
+const allModes = [ AM.read, AM.append, AM.create, AM.write, AM.delete ];
+
+// Based on https://github.com/solid/specification/issues/14#issuecomment-683480525
+// Columns: method, target, C/ permissions, C/R permissions, body, content-type, target exists, target does not exist
+// `undefined` implies C/R inherits the permissions of C/
+// For PUT/PATCH/DELETE we return 205 instead of 200/204
+/* eslint-disable no-multi-spaces */
+const table: [string, string, AM[], AM[] | undefined, string, string, number, number][] = [
+  // We currently handle OPTIONS before authorization
+  // [ 'OPTIONS', 'C/R', [],                     undefined,              '',     '',  401, 401 ],
+  // [ 'OPTIONS', 'C/R', [],                     [ AM.read ],            '',     '',  200, 404 ],
+  // [ 'OPTIONS', 'C/R', [ AM.read ],            undefined,              '',     '',  200, 404 ],
+
+  [ 'HEAD',    'C/R', [],                     undefined,              '',     '',  401, 401 ],
+  [ 'HEAD',    'C/R', [],                     [ AM.read ],            '',     '',  200, 404 ],
+  [ 'HEAD',    'C/R', [ AM.read ],            undefined,              '',     '',  200, 404 ],
+
+  [ 'GET',     'C/R', [],                     undefined,              '',     '',  401, 401 ],
+  [ 'GET',     'C/R', [],                     [ AM.read ],            '',     '',  200, 404 ],
+  [ 'GET',     'C/R', [ AM.read ],            undefined,              '',     '',  200, 404 ],
+  // For the 404, we only check if C/R has read permissions, not C/
+  // [ 'GET',     'C/R', [ AM.read ],            [ AM.write ],           '',     '',  401, 404 ],
```

Comment:
The idea is that disclosure of existence is a container matter (listing). So `404` is correct in the sense that the client is allowed to know about existence. (I don't think `401` is necessarily wrong, just more conservative.)

Reply:
Our current architecture does not allow us to check the parent permissions in this situation (without a lot of hassle), so I'm going to keep this on 401 for now then.

Reply:
Okay; why is it exactly though?

Reply:
Because the permission readers only expose the permissions on the targeted resource. See my comments here https://github.com/CommunitySolidServer/CommunitySolidServer/issues/1161#issuecomment-1054402389

But in summary we would have to a) always read the parent permissions for every request (instead of the create/delete cases we do now), and b) have the permission readers return permissions for multiple resources instead of just the target.

Reply:
> > always read the parent permissions for every request
> 
> …if the resource does not exist?

True.